### PR TITLE
[Validator] Document `PasswordStrength` constraint

### PR DESCRIPTION
**Pull request by Spomky**
These modifications are proposed as per ~https://github.com/symfony/symfony/pull/49789~ => https://github.com/symfony/symfony/pull/49856

### CODE REVIEW

File: reference/constraints/PasswordStrength.rst [hunk 1]
```diff
@@ -0,0 +1,103 @@
+PasswordStrength
+================
+
+Validates that the given password has reached the minimum strength required by
+the constraint. The strength is measured using the `zxcvbn-php`_. library.
```
Comment (MrYamous) [inline reference/constraints/PasswordStrength.rst]:
```suggestion
the constraint. The strength is measured using the `zxcvbn-php`_ library.
```

File: reference/constraints/PasswordStrength.rst [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            }
+        }
+
+Available Options
+-----------------
+
+.. include:: /reference/constraints/_groups-option.rst.inc
+
+``lowStrengthMessage``
+~~~~~~~~~~~
```
Comment (MrYamous) [inline reference/constraints/PasswordStrength.rst]:
```suggestion
~~~~~~~~~~~~~~~~~~~~~~
```

File: reference/constraints/PasswordStrength.rst [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+**type**: ``string`` **default**: ``The password strength is too low. Please use a stronger password.``
+
+The default message supplied when the password does not reach the minimum required score.
+
+.. include:: /reference/constraints/_payload-option.rst.inc
+
+``restrictedData``
+~~~~~~~~~~~~~~~
```
Comment (MrYamous) [inline reference/constraints/PasswordStrength.rst]:
```suggestion
~~~~~~~~~~~~~~~~~~
```

File: reference/constraints/PasswordStrength.rst [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+**type**: ``string[]`` **default**: ``[]``
+
+It is possible to determine if the submitted password contains restricted data
+such as the user given/family name,
+
+``restrictedDataMessage``
+~~~~~~~~~~~~~
```
Comment (MrYamous) [inline reference/constraints/PasswordStrength.rst]:
```suggestion
~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Review by MrYamous — COMMENTED** [review]
I think `minScore` option could be listed too in Available Options section

File: reference/constraints/PasswordStrength.rst (hunk 2 above)
Comment (OskarStark) [reply reference/constraints/PasswordStrength.rst]: @MrYamous looks like the Doctor-RST message is cut, it should be:

`Please ensure title "%s" and underline length are matching`

🤔

**Comment by OskarStark** [comment]
Reverted by @chalasr in https://github.com/symfony/symfony/pull/49831

**Comment by OskarStark** [comment]
The new PR was merged
* https://github.com/symfony/symfony/pull/49856

**Review by OskarStark — APPROVED** [review]

**Comment by OskarStark** [comment]
Thank you Florent.

**Pull request merged**